modules/neovim/neovim.nix

```diff
@@ -43,39 +79,23 @@
               ''
             );
           in
-          {
-            plugins = [
-              (lib.mkIf (cfg.plugin == "base16-nvim") {
-                plugin = pkgs.vimPlugins.base16-nvim;
-                type = "lua";
-                config = with config.lib.stylix.colors.withHashtag; ''
-                  require('base16-colorscheme').setup({
-                    base00 = '${base00}', base01 = '${base01}', base02 = '${base02}', base03 = '${base03}',
-                    base04 = '${base04}', base05 = '${base05}', base06 = '${base06}', base07 = '${base07}',
-                    base08 = '${base08}', base09 = '${base09}', base0A = '${base0A}', base0B = '${base0B}',
-                    base0C = '${base0C}', base0D = '${base0D}', base0E = '${base0E}', base0F = '${base0F}'
-                  })
-
-                  ${transparencyCfg}
-                '';
-              })
-              (lib.mkIf (cfg.plugin == "mini.base16") {
-                plugin = pkgs.vimPlugins.mini-nvim;
-                type = "lua";
-                config = with config.lib.stylix.colors.withHashtag; ''
-                  require('mini.base16').setup({
-                    palette = {
-                      base00 = '${base00}', base01 = '${base01}', base02 = '${base02}', base03 = '${base03}',
-                      base04 = '${base04}', base05 = '${base05}', base06 = '${base06}', base07 = '${base07}',
-                      base08 = '${base08}', base09 = '${base09}', base0A = '${base0A}', base0B = '${base0B}',
-                      base0C = '${base0C}', base0D = '${base0D}', base0E = '${base0E}', base0F = '${base0F}'
-                    }
-                  })
-
-                  ${transparencyCfg}
-                '';
-              })
-            ];
-          };
-      };
+          [
+            {
+              plugin =
+                if cfg.plugin == "mini.base16" then
+                  pkgs.vimPlugins.mini-nvim
+                else if cfg.plugin == "base16-nvim" then
+                  pkgs.vimPlugins.base16-nvim
+                else
+                  throw "stylix: neovim: plugin ${cfg.plugin} is not supported.";
```

Comment:
Although this arguably make the code more readable, I would prefer removing this since `cfg.plugins` should already error when it is not one of its allowed enum variants.